app/scripts/models/flow.js

```diff
@@ -0,0 +1,130 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A model to represent the somewhat nebulous concept of an
+ * "event flow". A single flow could be a user connecting a
+ * new device to their Sync account, for instance.
+ *
+ * Each flow has two properties.
+ *
+ *   * flowId: 32-byte identifier hex-encoded as a string.
+ *   * flowBeginTime: server-generated timestamp.
+ *
+ * When a new flow is begun, it should result in an activity
+ * event being logged by the server. To that end, we push a
+ * a `flow.begin` onto the metrics event array.
+ *
+ * There is some mess surrounding when to begin a new flow,
+ * which necessitates equivalent mess in this model. It runs
+ * something like this:
+ *
+ *   1. A user arriving at /signin or /signup constitutes the
+ *      beginning of a flow, so a corresponding `flow.begin`
+ *      event must be logged.
+ *
+ *   2. A user refreshing the page on /signin or /signup does
+ *      not constitute a new flow, so events must not be logged
+ *      in those cases. This is achieved by saving the flowId
+ *      in local storage.
```

Comment:
> count the number of users who try to initiate a device connection, rather than the number of attempts

*nod* an important distinction, and one that's not well called out in the spec; I think the number of attempts is closer to what we want to measure in this case, given the high-level goals in https://github.com/mozilla/fxa-content-server/pull/3619/files/8e499e41c3c5439ece942f24b67742e04a6e8018#r58148313